### PR TITLE
feat: Resolve Debug to redirect to UI tool on web

### PR DESCRIPTION
**Pull request by nickybondarenko**
This PR adds logging to DEBUG layer to redirect user to our web UI for Resolve debugging in case of anything but success resolve (MATCH).

### CODE REVIEW

File: Confidence/src/main/java/com/spotify/confidence/DebugLogger.kt [hunk 1]
```diff
@@ -35,6 +36,11 @@ internal class DebugLoggerImpl(private val filterLevel: LoggingLevel) : DebugLog
         verbose("[$action] $context")
     }
 
+    override fun logResolve(flag: String, context: Map<String, ConfidenceValue>) {
+        val trimContext = "$context".replace(" ", "")
+        debug("[Resolve Debug] https://app.confidence.spotify.com/flags/resolver-test?flag=flags/$flag&context=$trimContext")
```
Comment (nickybondarenko) [inline Confidence/src/main/java/com/spotify/confidence/DebugLogger.kt:41]: Currently, the context doesn't show up in the UI.

File: Confidence/src/main/java/com/spotify/confidence/DebugLogger.kt [hunk 2]
```diff
@@ -35,6 +36,11 @@ internal class DebugLoggerImpl(private val filterLevel: LoggingLevel) : DebugLog
         verbose("[$action] $context")
     }
 
+    override fun logResolve(flag: String, context: Map<String, ConfidenceValue>) {
+        val trimContext = "$context".replace(" ", "")
```
Comment (nicklasl) [inline Confidence/src/main/java/com/spotify/confidence/DebugLogger.kt:40]: not sure you want to do this. 
first; think about what format is expected. probably some json?
also, if we trim context values from `" "` then any context key or value that contains a `" "` would be incorrect, right?

File: Confidence/src/main/java/com/spotify/confidence/Confidence.kt [hunk 3]
```diff
@@ -106,16 +106,25 @@ class Confidence internal constructor(
...
+        val eval = cache.get().getEvaluation(
+            key,
+            default,
+            getContext()
+        ) { flagName, resolveToken ->
+            // this lambda will be invoked inside the evaluation process
+            // and only if the resolve reason is not targeting key error.
+            apply(flagName, resolveToken)
+        }
+        if (eval.reason !== ResolveReason.RESOLVE_REASON_MATCH) {
```
Comment (nicklasl) [inline Confidence/src/main/java/com/spotify/confidence/Confidence.kt:119]: I think we should do it even if we get a match.

File: Confidence/src/main/java/com/spotify/confidence/DebugLogger.kt (hunk 2 above)
Comment (nickybondarenko) [reply Confidence/src/main/java/com/spotify/confidence/DebugLogger.kt:40]: Right, I will figure out how to translate the context into a JSON..

**Review by nicklasl — DISMISSED** [review]

**Review by nicklasl — APPROVED** [review]

**Pull request merged**